plugins/net_plugin/include/eosio/net_plugin/auto_bp_peering.hpp

```diff
@@ -182,7 +182,7 @@ class bp_connection_manager {
 
                fc_dlog(self()->get_logger(), "pending_downstream_neighbors: ${pending_downstream_neighbors}",
                        ("pending_downstream_neighbors", to_string(pending_downstream_neighbors)));
-               for (auto neighbor : pending_downstream_neighbors) { self()->connections.connect(config.bp_peer_addresses[neighbor]); }
+               for (auto neighbor : pending_downstream_neighbors) { self()->connections.connect(config.bp_peer_addresses[neighbor], *self()->p2p_addresses.begin() ); }
```

Comment:
As discussed please update help and PR description indicating that the first configured server_address is reported in the handshake message to peers. Also guard against `p2p_addresses` being empty here.